internal/gengapic/paging.go

```diff
@@ -246,7 +269,21 @@ func (g *generator) pagingCall(servName string, m *descriptor.MethodDescriptorPr
 	p("  }")
 	p("")
 	p("  it.Response = resp")
-	p("  return resp.Get%s(), resp.GetNextPageToken(), nil", snakeToCamel(elemField.GetName()))
+	repeatedField, elems := fmt.Sprintf("resp.Get%s()", snakeToCamel(elemField.GetName())), ""
+	if pt.mapValueTypeName != "" {
```

Comment:
Is this the exact same code as what's in `genrest` `paginRESTCall`? Why is it needed in both? Should it be a function then?

Reply:
Along similar lines: paging.go is for gRPC-specific paging, or paging in general? And genrest.go is for REST in particular? It seems like some of the functionality/logic is common, so it could be factored out to prevent duplication.